api/v1alpha1/podhttpchaos_types.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2020 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// PodHttpChaosSpec defines the desired state of PodHttpChaos.
+type PodHttpChaosSpec struct {
+	// ProxyPorts represents the target ports to be proxy of.
+	ProxyPorts []int32 `json:"proxy_ports"`
+
+	// Rules are a list of injection rule for http request.
+	// +optional
+	Rules []*PodHttpChaosRule `json:"rules,omitempty"`
+}
+
+// PodHttpChaosStatus defines the actual state of PodHttpChaos.
+type PodHttpChaosStatus struct {
+	// Pid represents a running tproxy process id.
+	// +optional
+	Pid int64 `json:"pid,omitempty"`
```

Comment:
We should send the pid and startTime in protobuf, and it may be troublesome to use optional field in protobuf. Moreover, the zero value of both pid and start time is meaningless, so I think this is acceptable.